File: src/com/l2kt/gameserver/model/item/MercenaryTicket.java
package com.l2kt.gameserver.model.item;

import java.util.Arrays;

import com.l2kt.gameserver.instancemanager.SevenSigns.CabalType;
import com.l2kt.gameserver.templates.StatsSet;

public final class MercenaryTicket
{
	public enum TicketType
	{
		SWORD,
		POLE,
		BOW,
		CLERIC,
		WIZARD,
		TELEPORTER
	}
	
	private final int _itemId;
	private final TicketType _type;
	private final boolean _isStationary;
	private final int _npcId;
	private final int _maxAmount;
	private final CabalType[] _ssq;
	
	public MercenaryTicket(StatsSet set)
	{
		_itemId = set.getInteger("itemId");
		_type = set.getEnum("type", TicketType.class);
		_isStationary = set.getBool("stationary");
		_npcId = set.getInteger("npcId");
		_maxAmount = set.getInteger("maxAmount");
		
		final String[] ssq = set.getStringArray("ssq");
		
		_ssq = new CabalType[ssq.length];
		for (int i = 0; i < ssq.length; i++)
			_ssq[i] = Enum.valueOf(CabalType.class, ssq[i]);
	}
	
	public int getItemId()
	{
		return _itemId;
	}
	
	public TicketType getType()
	{
		return _type;
	}
	
	public boolean isStationary()
	{
		return _isStationary;
	}
	
	public int getNpcId()
	{
		return _npcId;
	}
	
	public int getMaxAmount()
	{
		return _maxAmount;
	}
	
	public boolean isSsqType(CabalType type)
	{
		return Arrays.asList(_ssq).contains(type);
	}
}

File: src/com/l2kt/gameserver/model/item/MercenaryTicket.kt
package com.l2kt.gameserver.model.item

import com.l2kt.gameserver.instancemanager.SevenSigns.CabalType
import com.l2kt.gameserver.templates.StatsSet

class MercenaryTicket(set: StatsSet) {

    val itemId: Int = set.getInteger("itemId")
    val type: TicketType
    val isStationary: Boolean
    val npcId: Int
    val maxAmount: Int
    private val _ssq: MutableList<CabalType>

    enum class TicketType {
        SWORD,
        POLE,
        BOW,
        CLERIC,
        WIZARD,
        TELEPORTER
    }

    init {
        type = set.getEnum("type", TicketType::class.java)
        isStationary = set.getBool("stationary")
        npcId = set.getInteger("npcId")
        maxAmount = set.getInteger("maxAmount")

        val ssq = set.getStringArray("ssq")

        _ssq = mutableListOf()
        for (i in ssq.indices)
            _ssq.add(i, CabalType.valueOf(ssq[i]))
    }

    fun isSsqType(type: CabalType): Boolean {
        return _ssq.contains(type)
    }
}